pkg/services/ngalert/store/alert_rule.go

```diff
@@ -419,21 +432,27 @@ func (st DBstore) UpdateRuleGroup(cmd UpdateRuleGroupCmd) error {
 				continue
 			}
 
+			new := ngmodels.AlertRule{
+				OrgID:           cmd.OrgID,
+				Title:           r.GrafanaManagedAlert.Title,
+				Condition:       r.GrafanaManagedAlert.Condition,
+				Data:            r.GrafanaManagedAlert.Data,
+				UID:             r.GrafanaManagedAlert.UID,
+				IntervalSeconds: int64(time.Duration(cmd.RuleGroupConfig.Interval).Seconds()),
+				NamespaceUID:    cmd.NamespaceUID,
+				RuleGroup:       ruleGroup,
+				NoDataState:     ngmodels.NoDataState(r.GrafanaManagedAlert.NoDataState),
+				ExecErrState:    ngmodels.ExecutionErrorState(r.GrafanaManagedAlert.ExecErrState),
+			}
+
+			if r.ApiRuleNode != nil {
```

Comment:
Is there a reason this object would be nil? I wondering if this should error instead.

Reply:
I believe it's using  sharing the `For`, `Annotations`, and `Labels` field from the `ApiRuleNode` rather than redefining the same fields itself.

Reply:
It can be nil when none of the above fields (`For`, `Annotations`, and `Labels`) is set like in [this](https://github.com/grafana/grafana/blob/d0b600e6e41dcb41fca9c73a98a7c12c1de97d33/pkg/services/ngalert/api/test-data/post-rulegroup-42.json#L79) example.